### ISSUE TITLE
Missing GPU implementation

### ISSUE THREAD

**Issue by kvanhoey** [tensorflow/tensorflow#32837]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Linux Kubuntu 18.04
- TensorFlow installed from: binary (with pip3)
- TensorFlow version: 1.14.0 (v1.14.0-rc1-22-gaf24dc91b5)
- Python version: 3.6.8
- CUDA/cuDNN version: 10.0.130
- GPU model and memory: NVidia Titan Xp

**Describe the current behavior**
The op tf.norm cannot be assigned to the GPU in all configurations (see minimal example below).

**Describe the expected behavior**
I see no theoretical reason for the tf.norm op not to be fully portable to the GPU. I suspect this is a bug, not an intended behavior.

**Code to reproduce the issue**
```
import tensorflow as tf

b = 10
width = 16
height = 16
channels = 3

def norm_simple(X):
  return tf.norm(X,axis=[1,2],ord=2)

def norm_custom(X):
  X_r = tf.reshape(X,shape=[b,width*height,channels])
  return tf.norm(X_r,axis=1,ord=2)

s = tf.Session()

A = tf.ones(shape=[b,width,height,channels])

with tf.device('/cpu:0'):
  n1_cpu = norm_simple(A)
s.run(n1_cpu)
print(" === norm_simple works on CPU")

with tf.device('/cpu:0'):
  n2_cpu = norm_custom(A)
s.run(n2_cpu)
print(" === norm_custom works on CPU")

with tf.device('/gpu:0'):
 n2_gpu = norm_custom(A)
s.run(n2_gpu)
print(" === norm_custom works on GPU")

with tf.device('/gpu:0'):
 n1_gpu = norm_simple(A)
s.run(n1_gpu)
print(" === norm_simple works on GPU")

```

**Other info / logs**
Execution of the above code on my machine:
```
 === norm_simple works on CPU
 === norm_custom works on CPU
 === norm_custom works on GPU
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1356, in _do_call
    return fn(*args)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1339, in _run_fn
    self._extend_graph()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1374, in _extend_graph
    tf_session.ExtendSession(self._session)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation norm_3/map/TensorArray: Could not satisfy explicit device specification '' because the node {{colocation_node norm_3/map/TensorArray}} was colocated with a group of nodes that required incompatible device '/device:GPU:0'. All available devices [/job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_GPU:0, /job:localhost/replica:0/task:0/device:XLA_GPU:1, /job:localhost/replica:0/task:0/device:XLA_CPU:0, /job:localhost/replica:0/task:0/device:GPU:0, /job:localhost/replica:0/task:0/device:GPU:1]. 
Colocation Debug Info:
Colocation group had the following types and supported devices: 
Root Member(assigned_device_name_index_=-1 requested_device_name_='/device:GPU:0' assigned_device_name_='' resource_device_name_='' supported_device_types_=[CPU, XLA_CPU, XLA_GPU] possible_devices_=[]
Switch: GPU CPU XLA_CPU XLA_GPU 
TensorArrayScatterV3: CPU XLA_CPU XLA_GPU 
TensorArrayReadV3: CPU XLA_CPU XLA_GPU 
Enter: GPU CPU XLA_CPU XLA_GPU 
TensorArrayV3: CPU XLA_CPU XLA_GPU 
Const: GPU CPU XLA_CPU XLA_GPU 

Colocation members, user-requested devices, and framework assigned devices, if any:
  norm_3/Const (Const) /device:GPU:0
  norm_3/map/TensorArray (TensorArrayV3) 
  norm_3/map/TensorArrayUnstack/TensorArrayScatter/TensorArrayScatterV3 (TensorArrayScatterV3) /device:GPU:0
  norm_3/map/while/TensorArrayReadV3/Enter (Enter) /device:GPU:0
  norm_3/map/while/TensorArrayReadV3 (TensorArrayReadV3) /device:GPU:0
  norm_3/map/while/cond/Switch_1 (Switch) /device:GPU:0
  norm_3/map/while/cond/add/Switch (Switch) /device:GPU:0

         [[{{node norm_3/map/TensorArray}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "bugreport.py", line 36, in <module>
    s.run(n1_gpu)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 950, in run
    run_metadata_ptr)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1173, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1350, in _do_run
    run_metadata)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1370, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation norm_3/map/TensorArray: Could not satisfy explicit device specification '' because the node node norm_3/map/TensorArray (defined at bugreport.py:9) placed on device No device assignments were active during op 'norm_3/map/TensorArray' creation.  was colocated with a group of nodes that required incompatible device '/device:GPU:0'. All available devices [/job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_GPU:0, /job:localhost/replica:0/task:0/device:XLA_GPU:1, /job:localhost/replica:0/task:0/device:XLA_CPU:0, /job:localhost/replica:0/task:0/device:GPU:0, /job:localhost/replica:0/task:0/device:GPU:1]. 
Colocation Debug Info:
Colocation group had the following types and supported devices: 
Root Member(assigned_device_name_index_=-1 requested_device_name_='/device:GPU:0' assigned_device_name_='' resource_device_name_='' supported_device_types_=[CPU, XLA_CPU, XLA_GPU] possible_devices_=[]
Switch: GPU CPU XLA_CPU XLA_GPU 
TensorArrayScatterV3: CPU XLA_CPU XLA_GPU 
TensorArrayReadV3: CPU XLA_CPU XLA_GPU 
Enter: GPU CPU XLA_CPU XLA_GPU 
TensorArrayV3: CPU XLA_CPU XLA_GPU 
Const: GPU CPU XLA_CPU XLA_GPU 

Colocation members, user-requested devices, and framework assigned devices, if any:
  norm_3/Const (Const) /device:GPU:0
  norm_3/map/TensorArray (TensorArrayV3) 
  norm_3/map/TensorArrayUnstack/TensorArrayScatter/TensorArrayScatterV3 (TensorArrayScatterV3) /device:GPU:0
  norm_3/map/while/TensorArrayReadV3/Enter (Enter) /device:GPU:0
  norm_3/map/while/TensorArrayReadV3 (TensorArrayReadV3) /device:GPU:0
  norm_3/map/while/cond/Switch_1 (Switch) /device:GPU:0
  norm_3/map/while/cond/add/Switch (Switch) /device:GPU:0

         [[node norm_3/map/TensorArray (defined at bugreport.py:9) ]]Additional information about colocations:No node-device colocations were active during op 'norm_3/map/TensorArray' creation.
No device assignments were active during op 'norm_3/map/TensorArray' creation.

Original stack trace for 'norm_3/map/TensorArray':
  File "bugreport.py", line 35, in <module>
    n1_gpu = norm_simple(A)
  File "bugreport.py", line 9, in norm_simple
    return tf.norm(X,axis=[1,2],ord=2)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/linalg_ops.py", line 600, in norm
    ops.convert_to_tensor(axis))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/map_fn.py", line 228, in map_fn
    for elem in elems_flat]
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/map_fn.py", line 228, in <listcomp>
    for elem in elems_flat]
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/tensor_array_ops.py", line 1086, in __init__
    name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/tensor_array_ops.py", line 164, in __init__
    self._handle, self._flow = create()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/tensor_array_ops.py", line 161, in create
    name=scope)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 8085, in tensor_array_v3
    tensor_array_name=tensor_array_name, name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py", line 3616, in create_op
    op_def=op_def)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py", line 2005, in __init__
    self._traceback = tf_stack.extract_stack()

```

**Comment by gadagashwini** [comment]
@kvanhoey, Thanks for reporting this issue.
I could reproduce the reported issue in TF1.x. Its fixed in TF 2.0.0rc2.
Please take a look at colab [gist](https://colab.sandbox.google.com/gist/gadagashwini/2ae2a31571acc201572fb9fa4e15e66c/untitled166.ipynb). You might want to give it a try instead. Thanks!

**Comment by kvanhoey** [comment]
Thank you for getting back to me. Glad it's fixed in 2.0.
I have a big project that I don't want to move to TF 2.0 yet, so I'll stay with my workaround (reshape first, then tf.norm on one dimension only).
